spark_client/services.py

```diff
@@ -870,3 +897,46 @@ def pyspark_shell(self, cli_property: Optional[str], extra_args: List[str]):
             self.logger.debug(submit_cmd)
             with environ(KUBECONFIG=self.kube_interface.kube_config_file):
                 os.system(submit_cmd)
+
+    def add_optional_detected_driver_host(self, conf: PropertyFile):
```

Comment:
Can you please add a documentation for this function and the other added functions